kms/google/cloud/kms_v1/gapic/key_management_service_client.py

```diff
@@ -0,0 +1,1762 @@
+                as the first argument and the default transport class as
+                the second argument.
+            channel (grpc.Channel): DEPRECATED. A ``Channel`` instance
+                through which to make calls. This argument is mutually exclusive
+                with ``credentials``; providing both will raise an exception.
+            credentials (google.auth.credentials.Credentials): The
+                authorization credentials to attach to requests. These
+                credentials identify this application to the service. If none
+                are specified, the client will attempt to ascertain the
+                credentials from the environment.
+                This argument is mutually exclusive with providing a
+                transport instance to ``transport``; doing so will raise
+                an exception.
+            client_config (dict): DEPRECATED. A dictionary of call options for
+                each method. If not specified, the default configuration is used.
+            client_info (google.api_core.gapic_v1.client_info.ClientInfo):
+                The client info used to send a user-agent string along with
+                API requests. If ``None``, then default info will be used.
+                Generally, you only need to set this if you're developing
+                your own client library.
+        """
+        # Raise deprecation warnings for things we want to go away.
+        if client_config:
+            warnings.warn('The `client_config` argument is deprecated.',
+                          PendingDeprecationWarning)
+        if channel:
+            warnings.warn(
+                'The `channel` argument is deprecated; use '
+                '`transport` instead.', PendingDeprecationWarning)
+
+        # Instantiate the transport.
+        # The transport is responsible for handling serialization and
+        # deserialization and actually sending data to the service.
+        if transport:
+            if callable(transport):
+                self.transport = transport(
+                    credentials=credentials,
+                    default_class=key_management_service_grpc_transport.
+                    KeyManagementServiceGrpcTransport,
+                )
+            else:
+                if credentials:
+                    raise ValueError(
+                        'Received both a transport instance and '
+                        'credentials; these are mutually exclusive.')
+                self.transport = transport
+        else:
+            self.transport = key_management_service_grpc_transport.KeyManagementServiceGrpcTransport(
+                address=self.SERVICE_ADDRESS,
+                channel=channel,
+                credentials=credentials,
+            )
+
+        if client_info is None:
+            client_info = (
+                google.api_core.gapic_v1.client_info.DEFAULT_CLIENT_INFO)
+        client_info.gapic_version = _GAPIC_LIBRARY_VERSION
```

Comment:
This bit is problematic -- the `DEFAULT_CLIENT_INFO` instance is going to be scribbled on by every library generated this way, which means that each time an app instantiates a generated client, they *all* use the last-instantiated one's `gapic_version`.

Rather than reuse the instance, we should be constructing a *new* one here, e.g.:

```python
        if client_info is None:  # create a new instance
            client_info = (
                google.api_core.gapic_v1.client_info.ClientInfo(
                    gapic_version=_GAPIC_LIBRARY_VERSION)
        else:  # update caller-supplied instance
            client_info.gapic_version = _GAPIC_LIBRARY_VERSION
```

Reply:
It seems we do this in released clients today just like this. I can make a bug on gapic-generator for it though.
Here is an example of us doing this:
https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/dlp/google/cloud/dlp_v2/gapic/dlp_service_client.py#L173

Reply:
https://github.com/googleapis/gapic-generator/issues/2161